Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/EmailService.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Net;
+using System.Net.Mail;
+using System.Linq;
+using Know_Your_Nation_Speedy.Models;
+
+
+namespace Know_Your_Nation_Speedy.Models{
+    public class EmailService
+    {
+        string smtpAddress = "smtp.gmail.com";
+        int portNumber = 587;
+        bool enableSSL = true;
+        string emailFromAddress = "**********"; //Sender Email Address  
+        string password = "***********"; //Sender Password   
+       
+        public bool SendMail(string To, string Subject, string Body)
+        {
+            try
+            {
+                using (MailMessage mail = new MailMessage())
+                {
+                    MailAssignment(mail, emailFromAddress, To, Subject, "<a href = 'http://ereader.retrotest.co.za/resetPassword'>Reset Password</h1>");
+                    SmtpSend(mail);
+                }
+            }
+            catch(Exception e)
+            {
+                Console.WriteLine(e.Message);
+                return false;
+            }
+            return true;
+        }
+
+        public void MailAssignment(MailMessage mailMessage, string From, string To, string Subject, string Body)
+        {
+            mailMessage.From = new MailAddress(From);
+            mailMessage.To.Add(To);
+            mailMessage.Subject = Subject;
+            mailMessage.IsBodyHtml = true;
+            mailMessage.Body = Body;
+        }
+
+        public void SmtpSend(MailMessage mail)
+        {
+            SmtpClient smtp = new SmtpClient(smtpAddress, portNumber);
+            smtp.Credentials = new NetworkCredential(emailFromAddress, password);
+            smtp.EnableSsl = enableSSL;
+            smtp.Send(mail);
+        }
+
+        public string generateCode()
```

Comment:
I dont think its necessary to write this whole function to generate a "unique" code, which, in fact, isn't unique. You can generate a GUID instead.

`GUID (or UUID) is an acronym for 'Globally Unique Identifier' (or 'Universally Unique Identifier'). It is a 128-bit integer number used to identify resources. The term GUID is generally used by developers working with Microsoft technologies, while UUID is used everywhere else.`

`128-bits is big enough and the generation algorithm is unique enough that if 1,000,000,000 GUIDs per second were generated for 1 year the probability of a duplicate would be only 50%. Or if every human on Earth generated 600,000,000 GUIDs there would only be a 50% probability of a duplicate.`